os/fs/driver/mtd/smart.c

```diff
@@ -4049,6 +4051,7 @@ static int smart_write_alloc_sector(FAR struct smart_struct_s *dev, uint16_t log
  *
  ****************************************************************************/
 
+#ifdef NXFUSE_HOST_BUILD
```

Comment:
you don't have to make 2 functions, I think.
You'd better modify smart_scan itself to apply extra logic for sector '0'

Reply:
HI Mr. Kwon,
The repeated definition is for the host-build case (NXFUSE_HOST_BUILD) where smartfs.bin image is being pre-built on the Linux(host) machine for flashing to the device. 
It separates the logic from what would be involved during on device smartfs operations.